docusaurus/docs/create-a-plugin/develop-a-plugin/work-with-data-frames.md

```diff
@@ -0,0 +1,138 @@
+---
+id: work-with-data-frames
+title: Work with data frames
+description: A guide to working with data frames in plugin development.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - data frames
+  - dataframes
+sidebar_position: 3
+---
+
+# Work with data frames
+
+The [data frame](../../introduction/data-frames.md) is a columnar data structure that allows for efficient querying of large amounts of data. Since data frames are a central concept when developing plugins for Grafana, in this guide we'll look at some ways you can use them.
+
+The `DataFrame` interface contains a `name` and an array of `fields` where each field contains the name, type, and the values for the field.
+
+:::note
+
+If you want to migrate an existing plugin to use the data frame format, refer to [Migrate to data frames](../../migration-guides/update-from-grafana-versions/v6.x-v7.x.md).
+
+:::
+
+## Create a data frame
+
+If you build a data source plugin, then you'll most likely want to convert a response from an external API to a data frame. Let's look at how to do this.
```

Comment:
I'd be tempted to remove this sentence. If we need to reference data source plugins we could do so in the intro

Reply:
done